test/uk/gov/hmrc/ddcopsusermanagmentfrontendpocscala/UserconnectorSpec.scala

```diff
@@ -14,66 +14,100 @@
  * limitations under the License.
  */
 
-import uk.gov.hmrc.http._
+package uk.gov.hmrc.ddcopsusermanagmentfrontendpocscala
+
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 import com.github.tomakehurst.wiremock.client.WireMock._
-import uk.gov.hmrc.http.test.{HttpClientSupport, WireMockSupport}
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
 import play.api.Configuration
 import play.api.libs.functional.syntax._
 import play.api.libs.json._
-import org.scalatest.matchers.should.Matchers
-import org.scalatest.wordspec.AnyWordSpec
-import org.scalatest.concurrent.ScalaFutures
-import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.Future
-
+import uk.gov.hmrc.http.HttpReads.Implicits._
+import uk.gov.hmrc.http._
+import uk.gov.hmrc.http.test.{HttpClientSupport, WireMockSupport}
 
+import scala.concurrent.{ExecutionContext, Future}
 
 case class User(title: String, firstName: String)
 
-class UserConnector(httpClient: HttpClient, configuration: Configuration) {
-    val baseUrl: String = configuration.get[String]("connector.url") //.getOrElse(throw new IllegalArgumentException("missing connector.url configuration"))
+class UserConnector(httpClient: HttpClient, configuration: Configuration)(implicit
+  executionContext: ExecutionContext
+) {
+  val baseUrl: String =
+    configuration.get[String](
+      "connector.url"
+    )
 
-    implicit val hc: HeaderCarrier = HeaderCarrier()
+  implicit val readsUser: Reads[User] = (
+    (JsPath \ "name" \ "title").read[String] and
+      (JsPath \ "name" \ "first").read[String]
+  )(User.apply _)
 
-    implicit val readsUser: Reads[User] = (
-        (__ \ "name" \ "title").read[String]
-        and (__ \ "name" \ "first").read[String]
-    )(User.apply _)
+  def search(query: String)(implicit hc: HeaderCarrier): Future[List[User]] =
+    httpClient
+      .GET[HttpResponse](
+        url = url"$baseUrl/search-user"
+      )
+      .map(response => Json.parse(response.body)("results").as[List[User]])
```

Comment:
@tszyrowski-MT it was something silly

```scala
    implicit val readsUsers: Reads[List[User]] = {
        (__ \ "results").read[List[User]]
    }
```

with that I was saying "here's how you read a list of users" but internally it tries to read a list of users again and that was messing stuff up. Instead I looked at the http-verbs docs and they provide a convenient way to just grab a representation of the response as a HttpResponse.